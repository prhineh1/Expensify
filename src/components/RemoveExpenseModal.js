import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = (props) => (
    <Modal 
        isOpen={props.isOpen}
        onRequestClose={props.toggleModal}
        contentLabel={'Remove Expense Warning'}
        closeTimeoutMS={200}
        className='modal'
    >
        <h3 className='modal__title'>Are you sure you want to remove this expense?</h3>
        <div className='modal__body'>
            <button className='button button--secondary' onClick={props.removeExpense}>Yes</button>
            <button className='button button--secondary' onClick={props.toggleModal}>No</button>
        </div>
    </Modal>
);

export default RemoveExpenseModal;