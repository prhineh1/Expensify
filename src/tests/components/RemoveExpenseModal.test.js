import React from 'react';
import { shallow } from 'enzyme';
import RemoveExpenseModal from '../../components/RemoveExpenseModal';

let isOpen, removeExpense, toggleModal, wrapper;

beforeEach(() => {
    isOpen = true;
    removeExpense = jest.fn();
    toggleModal = jest.fn()
    wrapper = shallow(<RemoveExpenseModal
        isOpen={isOpen}
        removeExpense={removeExpense}
        toggleModal={toggleModal}
     />);
});

test('should render RemoveExpensemodal', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle removeExpense', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(removeExpense).toHaveBeenCalled();
});

test('should handle toggleModal', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(toggleModal).toHaveBeenCalled();
});