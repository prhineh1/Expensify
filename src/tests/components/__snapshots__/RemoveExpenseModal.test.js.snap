// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpensemodal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Remove Expense Warning"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
>
  <h3
    className="modal__title"
  >
    Are you sure you want to remove this expense?
  </h3>
  <div
    className="modal__body"
  >
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;
